vm_insnhelper.c

```diff
@@ -3442,14 +3442,39 @@ vm_invoke_symbol_block(rb_execution_context_t *ec, rb_control_frame_t *reg_cfp,
 		       struct rb_calling_info *calling, const struct rb_callinfo *ci,
                        MAYBE_UNUSED(bool is_lambda), VALUE block_handler)
 {
-    VALUE val;
-    int argc;
-    VALUE symbol = VM_BH_TO_SYMBOL(block_handler);
-    CALLER_SETUP_ARG(ec->cfp, calling, ci);
-    argc = calling->argc;
-    val = vm_yield_with_symbol(ec, symbol, argc, STACK_ADDR_FROM_TOP(argc), calling->kw_splat, calling->block_handler);
-    POPN(argc);
-    return val;
+    int argc = calling->argc;
+
+    if (argc < 1) {
+        rb_raise(rb_eArgError, "no receiver given");
+    }
+    else if (argc > 1) {
+        /* E.g. when argc == 3
+         *
+         *   |      |        |      |  TOPN
+         *   |      |        +------+
+         *   |      |  +---> | arg1 |   -1
+         *   +------+  |     +------+
+         *   | arg1 | -+ +-> | arg0 |    0
+         *   +------+    |   +------+
+         *   | arg0 | ---+   |  BH  |    1
+         *   +------+        +------+
+         *   | recv |        | recv |    2
+         * --+------+--------+------+------
+         *
+         * INC_SP is done immediately below.
+         */
```

Comment:
Just a question: is it impossible to cause stack overflow?

Reply:
Maybe.  But this code is deleted later.

Reply:
https://github.com/ruby/ruby/pull/3152/commits/7ac31bbfc5afe3af2f77bc25bca4992f4c21b7d6